src/materialized/src/http/prof.rs

```diff
@@ -181,10 +161,37 @@ mod enabled {
         }
     }
 
+    fn flamegraph(stacks: StackProfile) -> anyhow::Result<Response<Body>> {
+        let collated = collate_stacks(stacks);
+        let data_json = RefCell::new(String::new());
+        collated.dfs(
+            |node| {
+                write!(
+                    data_json.borrow_mut(),
+                    "{{\"name\": \"{}\",\"value\":{},\"children\":[",
+                    node.name,
+                    node.weight
+                )
+                .unwrap(); // String's `std::fmt::Write` implementation never fails
+            },
+            |_node, is_last| {
+                data_json.borrow_mut().push_str("]}");
+                if !is_last {
+                    data_json.borrow_mut().push_str(",");
+                }
+            },
+        );
+        let data_json = &*data_json.borrow();
+        Ok(util::template_response(FlamegraphTemplate {
```

Comment:
Can you variablize the "Heap Flamegraph" bit of the template so that it says "CPU Flamegraph" or somesuch when rendering the CPU profile?